### PR TITLE
Fix IntegerNullable inference

### LINKED ISSUE #1550: Woodwork initialization fails with nullable int in dataframe
If you try to initialize woodwork on a dataframe with nullable integers it correctly infers the type but throws an error when it uses the pandas type conversion on the underlying data (because pandas turns anything with nans into floats)

The error is:

```
    def transform(self, series, null_invalid_values=False):
        """Converts the series dtype to match the logical type's if it is different."""
        new_dtype = self._get_valid_dtype(type(series))
        if new_dtype != str(series.dtype):
            # Update the underlying series
            try:
                series = series.astype(new_dtype)
            except (TypeError, ValueError):
>               raise TypeConversionError(series, new_dtype, type(self))
E               woodwork.exceptions.TypeConversionError: Error converting datatype for b from type float64 to type Int64. Please confirm the underlying data is consistent with logical type IntegerNullable.

/usr/local/lib/python3.9/site-packages/woodwork/logical_types.py:76: TypeConversionError
```

#### Code Sample, a copy-pastable example to reproduce your bug.

```python
        data = DataFrame.from_dict(
            {
                "a": [numpy.inf, numpy.nan, numpy.NAN, -numpy.Inf, None],
                "b": [numpy.finfo("d").max, numpy.finfo("d").min, 3, 1, None],
            }
        )
        data = data.ww.init()
```

### PR DESCRIPTION
**Pull request by bchen1116**
fix #1550

### CODE REVIEW

**Comment by bchen1116** [comment]
Results of initialization with too-large values:
image

File: woodwork/tests/type_system/test_ltype_inference.py [hunk 1]
```diff
@@ -204,6 +204,8 @@ def test_categorical_integers_inference(integers):
             dtypes = get_spark_dtypes(dtypes)
         for series in integers:
             for dtype in dtypes:
+                if len(integers) < 12 and "64" not in dtypes:
```
Comment (bchen1116) [inline woodwork/tests/type_system/test_ltype_inference.py:207]: values too large for other dtypes

File: woodwork/tests/type_system/test_ltype_inference.py [hunk 2]
```diff
@@ -215,7 +215,10 @@ def test_categorical_double_inference(doubles):
             dtypes = get_spark_dtypes(dtypes)
         for series in doubles:
             for dtype in dtypes:
+                # if len(doubles) < 16 and dtypes == "float_":
+                #     continue
```
Comment (ParthivNaresh) [inline woodwork/tests/type_system/test_ltype_inference.py:219]: No longer necessary prints?

**Review by ParthivNaresh — APPROVED** [review]
Looks good! Quick question, we don't need to change the inference function for `Integer` or `Double` right? This isn't something they're affected by?

**Comment by bchen1116** [comment]
> Looks good! Quick question, we don't need to change the inference function for `Integer` or `Double` right? This isn't something they're affected by?

The `Integer` inference function uses the `IntegerNullable` inference, so we should be good!

**Comment by ParthivNaresh** [comment]
> > Looks good! Quick question, we don't need to change the inference function for `Integer` or `Double` right? This isn't something they're affected by?
> 
> The `Integer` inference function uses the `IntegerNullable` inference, so we should be good!

Ah and an otherwise integer series with an `np.inf` becomes `float64`, so that works out to the same so it won't belong to the integer function either way, nice!

**Pull request merged**